cmake/templates/setup.bat.in

```diff
@@ -42,10 +42,22 @@ call %_SETUP_TMP%
 del %_SETUP_TMP%
 
 REM source all environment hooks
-for /F "tokens=* delims=;" %%a in ("%_CATKIN_ENVIRONMENT_HOOKS%") do (
-  for %%b in (%%a) do (
-    call "%%b"
-  )
+set _HOOK_COUNT=0
+:hook_loop
+IF %_HOOK_COUNT% LSS %_CATKIN_ENVIRONMENT_HOOKS_COUNT% (
+  REM set workspace for environment hook
+  call set CATKIN_ENV_HOOK_WORKSPACE=%%_CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%_WORKSPACE%%
+  call set _CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%_WORKSPACE=
+
+  REM call environment hook
+  call %%_CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%%%
+  call set _CATKIN_ENVIRONMENT_HOOKS_%_HOOK_COUNT%%=
+  
+  REM Cleanup local variables
+  set CATKIN_ENV_HOOK_WORKSPACE=
```

Comment:
A temporary variable should start with an `_` to avoid colliding with environment variable the user might have set.